xbatcher/generators.py

```diff
@@ -90,6 +98,10 @@ class BatchGenerator:
     preload_batch : bool, optional
         If ``True``, each batch will be loaded into memory before reshaping /
         processing, triggering any dask arrays to be computed.
+    squeeze_batch_dim : bool, optional
+        If ``False`` and all dims are input dims, each batch's dataset will have a
+        "batch" dimension of size 1 prepended to the array. This functionality is
```

Comment:
Some suggestions on the documentation. Maybe L86 and L87 could be edited to indicate the squeeze behavior is controllable with `squeeze_batch_dim`?

Also, this sentence might be a bit confusing to read. So an extra dimension of size 1 is added/prepended only when `batch_dims is None` or unset. For cases where `len(batch_dims) ＞= 1)`, the squeezing/collapsing of dimensions is still happening though. I'm wondering what's a good way to reword this to make it clearer on what is happening and why this option exists.

Reply:
The problem would only appear in this one corner case, as far as I could tell. So, my solution was coded to only apply when there were no `batch_dims` (at least that was my intention). If you're changing the default behavior though, you can probably make this simpler, and therefore the docs would be less confusing too.